api/sms_provider/sms_provider_test.go

```diff
@@ -0,0 +1,216 @@
+package sms_provider
+
+import (
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"net/url"
+	"testing"
+
+	"github.com/netlify/gotrue/conf"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"gopkg.in/h2non/gock.v1"
+)
+
+var handleApiRequest func(*http.Request) (*http.Response, error)
+
+type SmsProviderTestSuite struct {
+	suite.Suite
+	Config *conf.GlobalConfiguration
+}
+
+type MockHttpClient struct {
+	mock.Mock
+}
+
+func (m *MockHttpClient) Do(req *http.Request) (*http.Response, error) {
+	return handleApiRequest(req)
+}
+
+func TestSmsProvider(t *testing.T) {
+	ts := &SmsProviderTestSuite{
+		Config: &conf.GlobalConfiguration{
+			Sms: conf.SmsProviderConfiguration{
+				Twilio: conf.TwilioProviderConfiguration{
+					AccountSid:        "test_account_sid",
+					AuthToken:         "test_auth_token",
+					MessageServiceSid: "test_message_service_id",
+				},
+				Messagebird: conf.MessagebirdProviderConfiguration{
+					AccessKey:  "test_access_key",
+					Originator: "test_originator",
+				},
+				Vonage: conf.VonageProviderConfiguration{
+					ApiKey:    "test_api_key",
+					ApiSecret: "test_api_secret",
+					From:      "test_from",
+				},
+				Textlocal: conf.TextlocalProviderConfiguration{
+					ApiKey: "test_api_key",
+					Sender: "test_sender",
+				},
+			},
+		},
+	}
+	suite.Run(t, ts)
+}
+
+func (ts *SmsProviderTestSuite) TestTwilioSendSms() {
+	defer gock.Off()
```

Comment:
I usually use `defer gock.OffAll()` so that unmatched requests don't carry over to the next test. https://pkg.go.dev/github.com/h2non/gock#OffAll